### ISSUE TITLE
`isinstance(variable, AGenericType)` causes variable to be `AGenericType[unknown]`

### ISSUE THREAD

**Issue by cleoold** [microsoft/pyright#2695]
**Describe the bug**
`isinstance` typeguard cause the variable of a generic type to lose its type parameter

**To Reproduce**
Use pyright vscode and paste the code below.

**Expected behavior**
`value` should be `Sequence[T]` as in earlier version

**Screenshots or Code**
```py
from typing import Iterable, Sequence, TypeVar

T = TypeVar('T')

def fun(value: Iterable[T]):
    value  # Iterable[T@fun]
    if isinstance(value, Sequence):
        value  # Sequence[unknown]
```

**VS Code extension or command-line**
VSCode extension v1.1.194.

**Additional context**

**Comment by erictraut** [comment]
You mentioned "as in earlier version". Does that mean that you think this is a regression from an earlier version of pyright? I went back over 30 versions to see if I could find a point when this behavior may have changed, but I think it has always worked this way.

For what it's worth, mypy narrows to `Sequence[Any]` in this case.

In any case, this will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.195, which I just published. It will also be included in the next release of pylance.

**Comment by cleoold** [comment]
Okay. It's my bad if this wasn't a regression. When I saw this in latest
version it seemed to get mixed with other type errors.
